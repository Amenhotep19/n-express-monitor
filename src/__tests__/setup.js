import * as nAutoMetrics from '@financial-times/n-auto-metrics';
import * as nAutoLogger from '@financial-times/n-auto-logger';

import setupMonitor from '../setup';

jest.mock('@financial-times/n-auto-metrics');
jest.mock('@financial-times/n-auto-logger');

describe('setupMonitor', () => {
	afterEach(() => {
		jest.resetAllMocks();
	});

	afterAll(() => {
		jest.resetModules();
	});

	const app = {
		use: jest.fn(),
	};
	const metrics = {};
	const logger = {};

	it('initAutoMetrics if metrics instance is provided', () => {
		setupMonitor({ app, metrics });
		expect(nAutoMetrics.initAutoMetrics).toBeCalledWith(metrics);
	});

	it('setupLoggerInstance to override the default if logger is provided', () => {
		setupMonitor({ app, logger });
		expect(nAutoLogger.setupLoggerInstance).toBeCalledWith(logger);
	});

	it('do include enhancedRender middleware with autoNext default to true', () => {
		setupMonitor({ app });
		expect(app.use.mock.calls).toMatchSnapshot();
	});

	it('do not include enhancedRender middleware is autoNext disabled', () => {
		setupMonitor({ app, autoNext: false });
		expect(app.use.mock.calls).toMatchSnapshot();
	});
});
