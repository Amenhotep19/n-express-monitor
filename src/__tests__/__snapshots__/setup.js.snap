// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupMonitor do include enhancedRender middleware with autoNext default to true 1`] = `
Array [
  Array [
    /\\^\\\\/\\(\\?!_\\{2\\}\\)\\.\\*\\$/,
    Array [
      [MockFunction],
      [MockFunction],
    ],
  ],
]
`;

exports[`setupMonitor do not include enhancedRender middleware is autoNext disabled 1`] = `
Array [
  Array [
    /\\^\\\\/\\(\\?!_\\{2\\}\\)\\.\\*\\$/,
    Array [
      [MockFunction],
    ],
  ],
]
`;
